plugin.video.viaplay/resources/lib/viaplay.py

```diff
@@ -0,0 +1,347 @@
+﻿# -*- coding: utf-8 -*-
+"""
+A Kodi-agnostic library for Viaplay
+"""
+import codecs
+import os
+import cookielib
+import calendar
+import re
+import json
+import uuid
+import HTMLParser
+from collections import OrderedDict
+from datetime import datetime, timedelta
+
+import iso8601
+import requests
+
+
+class Viaplay(object):
+    def __init__(self, settings_folder, country, debug=False):
+        self.debug = debug
+        self.country = country
+        self.settings_folder = settings_folder
+        self.cookie_jar = cookielib.LWPCookieJar(os.path.join(self.settings_folder, 'cookie_file'))
+        self.tempdir = os.path.join(settings_folder, 'tmp')
+        if not os.path.exists(self.tempdir):
+            os.makedirs(self.tempdir)
+        self.deviceid_file = os.path.join(settings_folder, 'deviceId')
+        self.http_session = requests.Session()
+        self.device_key = 'xdk-%s' % self.country
+        self.base_url = 'https://content.viaplay.{0}/{1}'.format(self.country, self.device_key)
+        self.login_api = 'https://login.viaplay.%s/api' % self.country
+        try:
+            self.cookie_jar.load(ignore_discard=True, ignore_expires=True)
+        except IOError:
+            pass
+        self.http_session.cookies = self.cookie_jar
+
+    class ViaplayError(Exception):
+        def __init__(self, value):
+            self.value = value
+
+        def __str__(self):
+            return repr(self.value)
+
+    def log(self, string):
+        if self.debug:
+            try:
+                print '[Viaplay]: %s' % string
+            except UnicodeEncodeError:
+                # we can't anticipate everything in unicode they might throw at
+                # us, but we can handle a simple BOM
+                bom = unicode(codecs.BOM_UTF8, 'utf8')
+                print '[Viaplay]: %s' % string.replace(bom, '')
```

Comment:
`print('[Viaplay]: %s' % string.replace(bom, ''))`

Reply:
@Rechi Would you mind if I fix this in upcoming versions? I'm sure there is a lot more to take care of in regards to Python 3 compatibility.

Reply:
IIRC using `print` (no matter, Python 2 or 3 style) is not recommended for logging. All logging should be done with `xbmc.log()`

Reply:
Correct. Except for unit testing that would operate outside Kodi. 
If @emilsvennesson can update the PR or will do follow up to fix remaining items in a follow up PR i'd be ok with that to not stall the update too long.

Reply:
@romanvm @MartijnKaijser `viaplay.py` is supposed to be independent of Kodi, which is why I'm using `print` rather than importing any of the xbmc modules.

Reply:
Ok, but you should also use:
```python
from __future__ import print_function
```

Reply:
@emilsvennesson you should use the xbmc.log method instead of the print statement

Reply:
from my side updating it isn't a hard requirement, just wanted to inform you that this won't work any more with python3